### PR TITLE
Update visibility-related apis

### PR DESCRIPTION
**Pull request by ry323**

### CODE REVIEW

File: ClassTranscribeServer/Controllers/MediaController.cs [hunk 1]
```diff
@@ -58,6 +58,16 @@ public async Task<ActionResult<MediaDTO>> GetMedia(string id)
                 }
             }
 
+            var visibilityAuthorizationResult = await _authorizationService.AuthorizeAsync(User, media.Playlist.Offering, Globals.POLICY_UPDATE_OFFERING);
+            if (!visibilityAuthorizationResult.Succeeded)
+            {
+                if (media.Visibility == Visibility.Hidden || media.Playlist.Visibility == Visibility.Hidden || media.Playlist.Offering.Visibility == Visibility.Hidden)
```
Comment (chirantan31) [inline ClassTranscribeServer/Controllers/MediaController.cs:64]: This line could throw an error, if it does. The problem should be that the related objects have to be loaded. (i.e. media.Playlist and media.Playlist.Visibility) 

You'll have to do something like this https://github.com/classtranscribe/WebAPI/blob/a117f169872cbef1ef6ff142d38b5160a2295804/TaskEngine/Tasks/ProcessVideoTask.cs#L30 

Only if it fails.

Comment (ry323) [reply ClassTranscribeServer/Controllers/MediaController.cs:64]: I've tested those api and fixed some bugs. Here the loading of [IgnoreDataMember] works well without 'Include' so I didn't make any changes.

File: ClassTranscribeServer/Controllers/OfferingsController.cs [hunk 2]
```diff
@@ -36,7 +36,7 @@ public class OfferingsController : BaseController
         public async Task<ActionResult<IEnumerable<OfferingDTO>>> GetOfferingsByStudent()
         {
             // Store all visible offerings
-            List<Offering> offerings = await _context.Offerings.Where(o.Visibility == Visibility.Visible).ToListAsync();
+            List<Offering> offerings = await _context.Offerings.Where(o => o.Visibility == Visibility.Visible).ToListAsync();
```
Comment (robkooper) [inline ClassTranscribeServer/Controllers/OfferingsController.cs:39]: surprised this even compiled.

**Comment by robkooper** [comment]
were there actual compile errors, or did I read that wrong? If there are we should really start to look at how to use github actions to compile.

This might be a nice way to do this: https://github.com/actions/setup-dotnet

**Pull request closed without merging**